btsolver.py

```diff
@@ -122,10 +122,16 @@ def nakedTriples(self):
         return self.naked(length=3)
 
     def forwardChecking(self):
-        """
-           TODO:  Implement forward checking.
-        """
-        pass
+        for v in self.network.variables:
+            if v.isAssigned() and v.isModified():
+                for u in self.network.getNeighborsOfVariable(v):
+                    if u.getAssignment() == v.getAssignment():
+                        return False
+                    if not u.isAssigned():
+                        u.removeValueFromDomain(v.getAssignment())
+                        if u.size() == 0:
+                            return False
+        return True
```

Comment:
doesn't forward checking only need to check 1 variable and it's neighbors? looks like your iterating over all of them. don't make any changes though, i'll merge it with some stuff I have already